lib/internal/webidl.js

```diff
@@ -263,6 +264,74 @@ function type(V) {
   }
 }
 
+// https://webidl.spec.whatwg.org/#js-dictionary
+function createDictionaryConverter(members) {
+  // The spec requires us to operate the members of a dictionary in
+  // lexicographical order. We are doing all of these in the outer scope to
+  // reduce the overhead that could happen in the returned function.
+  const sortedMembers = [];
+  for (let i = 0; i < members.length; i++) {
+    const member = members[i];
+    ArrayPrototypePush(sortedMembers, member);
+  }
+
+  ArrayPrototypeSort(sortedMembers, (a, b) => {
+    if (a.key === b.key) {
+      return 0;
+    }
+    return a.key < b.key ? -1 : 1;
+  });
+
+  return function(
+    V,
+    opts = kEmptyObject,
+  ) {
+    const typeV = type(V);
+    if (typeV !== 'Undefined' && typeV !== 'Null' && typeV !== 'Object') {
```

Comment:
Why do you nedd to compare as a string? This is extremely slow. Spec compliance doesn't mean that we should implement it 1 by 1.

Reply:
Good point! Updated.